store/root/store_test.go

```diff
@@ -89,11 +89,45 @@ func (s *RootStoreTestSuite) TestQuery() {
 	// ensure the proof is non-nil for the corresponding version
 	result, err := s.rootStore.Query(defaultStoreKey, 1, []byte("foo"), true)
 	s.Require().NoError(err)
-	s.Require().NotNil(result.Proof.Proof)
-	s.Require().Equal([]byte("foo"), result.Proof.Proof.GetExist().Key)
-	s.Require().Equal([]byte("bar"), result.Proof.Proof.GetExist().Value)
+	s.Require().NotNil(result.ProofOps)
+	s.Require().Equal([]byte("foo"), result.ProofOps[0].Key)
 }
 
+// func (s *RootStoreTestSuite) TestQueryProof() {
+// 	// store1
+// 	bs1 := s.rootStore.GetBranchedKVStore("store1")
```

Comment:
@alexanderbez `BranchedKVStore` is weird, there is no `storeKey` field in `Set` function, how to manage the storeKey in rootStore?

Reply:
A `BranchKV` is already scoped to a StoreKey -- there's no need to have StoreKey be an argument to any APIs on it.

E.g. `s.rootStore.GetBranchedKVStore("store1")` returns a BranchKV scoped to `store1`.

What are you trying to achieve exactly with `RootStore`?

Reply:
> E.g. s.rootStore.GetBranchedKVStore("store1") returns a BranchKV scoped to store1.

no, it returns `rootStore` which is not related to `store1` key

```
// GetKVStore returns the store's root KVStore. Any writes to this store without
// branching will be committed to SC and SS upon Commit(). Branching will create
// a branched KVStore that allow writes to be discarded and propagated to the
// root KVStore using Write().
func (s *Store) GetKVStore(_ string) store.KVStore {
	if s.TracingEnabled() {
		return trace.New(s.rootKVStore, s.traceWriter, s.traceContext)
	}

	return s.rootKVStore
}

func (s *Store) GetBranchedKVStore(_ string) store.BranchedKVStore {
	if s.TracingEnabled() {
		return trace.New(s.rootKVStore, s.traceWriter, s.traceContext)
	}

	return s.rootKVStore
}
```

Reply:
Ahhh, yes yes! I forgot, this is when we had a RootStore with a single default store key. Then we created the SC abstraction that can support multi-store-keys (storekey per SC Tree).

But in the RootStore, we still only have a single KVStore. There's technically not a reason to have multiple KVStores per store key in RootStore, because the store service does prefixing for us.

Reply:
yes, tbh I am not sure how to interact with storeKey in rootStore

Reply:
also I can't see any prefixing

Reply:
It seems we need a `KVStore` per `StoreKey`. I'll make a PR to unblock you 👍

Reply:
https://github.com/cosmos/cosmos-sdk/pull/18968